decidim-proposals/app/controllers/concerns/decidim/proposals/needs_permission.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Decidim
+  module Proposals
+    # Common logic to ordering resources
+    module NeedsPermission
+      extend ActiveSupport::Concern
+
+      included do
+        helper_method :allowed_to?
+
+        class ::Decidim::MissingPermission < StandardError
+        end
+
+        rescue_from Decidim::MissingPermission, with: :user_not_authorized
+
+        def check_permission_to(action, subject, extra_context = {})
```

Comment:
The method name is misleading. `check` sounds passive (return either true or false) but this is `enforcing` or `ensuring` permissions instead. It's more evident when you put it with the previous `authorize!` method side-by-side.

Reply:
Done! It's now called `enforce_permission_to`